### PR TITLE
[Backend] - Read Cart API

### PR DESCRIPTION
**Pull request by sandrex-SGV**
## Ticket/s

- [[API] Read Cart](https://trello.com/c/V8GQsagG/38-api-read-cart)

## Changes

- add read cart API

## Screenshot

- Cart is empty

image

- User has Cart

image

## How to test?

- open Postman
- login, get bearer token and update request header.
- add product to cart, make sure product exists in db. use cURL below:
```
curl --location 'localhost:3001/api/cart' \
--header 'Authorization: Bearer [insert token here]' \
--header 'Content-Type: application/json' \
--data '{
    "productID": "[insert product id here]",
    "quantity": 10
}'
```
- read cart, use cURL below:
```
curl --location 'localhost:3001/api/cart/64ccca74b398b805cbbd942c' \
--header 'Authorization: Bearer [insert token here]'
```

## Notes

- if userID has existing cart, returns the cart object, otherwise returns message ''Cart is empty"

### CODE REVIEW

File: apps/api/src/cart/cart.controller.ts [hunk 1]
```diff
@@ -30,6 +31,13 @@ export class CartController {
     return cart;
   }
 
+  @Get('/:id')
+  async getCart(@Param('id') userID: string) {
+    const userCart = await this.cartService.getCart(userID);
+    if (!userCart) throw new NotFoundException('Cart is empty');
+    return userCart;
+  }
+
```
Comment (brentmillen) [inline apps/api/src/cart/cart.controller.ts]: utilize our access token to identify user. no need to pass id as param.

**Review by brentmillen — CHANGES_REQUESTED** [review]

File: apps/api/src/cart/cart.controller.ts (hunk 1 above)
Comment (sandrex-SGV) [reply apps/api/src/cart/cart.controller.ts]: @brentmillen resolved sir please check recent changes. thanks

**Review by TreshAnn — APPROVED** [review]
everything is working as expected on my end

**Review by brentmillen — APPROVED** [review]

File: apps/api/src/cart/cart.controller.ts [hunk 2]
```diff
@@ -30,6 +31,14 @@ export class CartController {
     return cart;
   }
 
+  @Get('/')
+  async getCart(@Request() req) {
+    const userID = await this.cartService.extractIdFromToken(req);
+    const userCart = await this.cartService.getCart(userID);
+    if (!userCart) throw new NotFoundException('Cart is empty');
```
Comment (brentmillen) [inline apps/api/src/cart/cart.controller.ts:37]: move the condition for the exception in service to allow handling of this error when the function is re-used.

**Review by brentmillen — DISMISSED** [review]

**Review by brentmillen — APPROVED** [review]

**Pull request merged**